src/Microsoft.AspNet.Http.Features/FeatureCollection.cs

```diff
@@ -93,6 +94,21 @@ IEnumerator IEnumerable.GetEnumerator()
             }
         }
 
+        public TFeature Get<TFeature>()
+        {
+            var type = typeof(TFeature);
+            if (type.GetTypeInfo().IsValueType)
```

Comment:
`(TFeature)this[type]` blows up if its a struct and not in the collection with a NRE